doc/src/specifications/platforms/README.md

```diff
@@ -1,19 +1,51 @@
+# Platforms
+
+A "platform" is the client-side context in which a user interacts with an app published in a psibase package. Some example platforms could include the web, a CLI program, a game console, etc.
+
 # Platform independence
 
-The only part of a psibase app that is not cross-platform is the user interface. Because psibase apps have an architectural separation of the cross-platform component (plugin) from the platform-native component (user interface), the user interface components are smaller/lighter than they would be otherwise.
+Most of the layers in the psibase app technology stack are platform independent. The part of a typical psibase app include:
+1. Service
+1. RPC/Http handlers
+1. Plugin
+1. User interface
+
+Of these, the user-interface is the only part of the app that is specific to the platform. The service and rpc/http handlers run server-side (so are not relevant to the platform), and the plugin, though it runs client side, is compiled to wasm and runs in a platform-provided WebAssembly host.
+
+Therefore, in the client-side portion of a psibase app, the cross-platform plugin is separated from the platform-native user interface. This separation allows for functionality to be reused between platforms, making the user-interfaces smaller/lighter than they would be otherwise.
+
+# Platform-specific app requirements
+
+The only part of a psibase app that is not cross-platform is the user interface. If a package published on a psibase network is intended to interface directly with a user, then it must deploy a user-interface that meets the specific requirements of the platform(s) on which it should be usable.
```

Comment:
This paragraph feels pretty clunky to me. The first sentence is also basically the same as two paragraphs up.

Reply:
Updated